File: CsChatClient/Messages/MediaMessageContent.cs
namespace CsChatClient.Messages
{
    public abstract class MediaMessageContent : MessageContent
    {
        public string LocalPath { get; set; }
        public string RemoteUrl { get; set; }
        public MediaType MediaType { get; set; }

        override
        public MessagePayload Encode()
        {
            MessagePayload payload = new MessagePayload();
            payload.LocalMediaPath = LocalPath;
            payload.RemoteMediaUrl = RemoteUrl;
            payload.MediaType = MediaType;
            return payload;
        }

        override
        public void Decode(MessagePayload payload)
        {
            LocalPath = payload.LocalMediaPath;
            RemoteUrl = payload.RemoteMediaUrl;
            MediaType = payload.MediaType;
        }
    }
}

File: CppChatClient/message/MediaMessageContent.cpp
//
//  MediaMessageContent.cpp
//  ChatClient
//
//  Created by Hao Jia on 2020/1/31.
//  Copyright © 2020 WF Chat. All rights reserved.
//

#include "MediaMessageContent.h"

namespace WFCLib {

MessagePayload MediaMessageContent::encode() const {
    MessagePayload payload = MessageContent::encode();
    payload.localMediaPath = localPath;
    payload.remoteMediaUrl = remoteUrl;
    payload.mediaType = getMediaType();
    return payload;
}

void MediaMessageContent::decode(const MessagePayload &payload) {
    MessageContent::decode(payload);
    localPath = payload.localMediaPath;
    remoteUrl = payload.remoteMediaUrl;
}
};
